### DISCUSSION TITLE
TS2300 [ERROR]: Duplicate identifier 'Deno' in Deno example

### DISCUSSION THREAD

**Discussion by johanbrook** [nanojsx/nano#63] [Help / Questions] [answered]
Hi! Thanks for a great module.

I'm writing a small Deno app, and I'm getting this when doing `deno run`:

```
TS2300 [ERROR]: Duplicate identifier 'Deno'.
declare var Deno: any
            ~~~~
    at https://deno.land/x/nano_jsx@v0.0.24/core.types.ts:12:13

    'Deno' was also declared here.
    declare namespace Deno {
                      ~~~~
        at asset:///lib.deno.ns.d.ts:67:19    and here.
    declare namespace Deno {
                      ~~~~
        at asset:///lib.deno.unstable.d.ts:6:19    and here.
    declare namespace Deno {
                      ~~~~
        at asset:///lib.deno.net.d.ts:6:19
```

My Deno config is as this:

```jsonc
{
    "compilerOptions": {
        "lib": ["dom", "dom.iterable", "dom.asynciterable", "deno.ns", "deno.unstable"],
        "jsxFactory": "h",
        // NanoJSX needs this
        "strictPropertyInitialization": false
    }
}
```

```bash
$ deno --version
deno 1.16.2 (release, x86_64-apple-darwin)
v8 9.7.106.2
typescript 4.4.2
```

I've tried to follow https://github.com/nanojsx/nano-jsx-deno-example, but not sure I've missed something? Why does NanoJSX redeclare the `Deno` var in its types at https://deno.land/x/nano_jsx@v0.0.24/core.types.ts?

Thanks!

**Comment by yandeu** [comment]
Sorry for that. It is hard to build an isomorphic library that works in Node.js, Deno and the Browser.
Please use [`--no-check`](https://deno.land/manual@v1.4.1/getting_started/typescript#code--no-checkcode-option) for now.

**Reply by yandeu** [reply]
Please try v25

**Reply by johanbrook** [reply]
v25 works! Thanks so much.

**Comment by johanbrook** [comment, marked as answer]
Works in v25 now.
